### DISCUSSION TITLE
mRootNode transformation?

### DISCUSSION THREAD

**Discussion by alessandroAM** [assimp/assimp#3812] [Q&A]
I have a very simple FBX, no bones, nodes, 1 only mesh, but I can't apply any transformation to it.

```
const AiScene *importedScene = importer.ReadFile(somePath, aiProcessPreset_TargetRealtime_Quality | aiProcess_Triangulate);

aiVector3D scaling = aiVector3D(10, 10, 10);

aiQuaternion rotation = aiQuaternion(aiVector3D(0, 1, 1), 45.0f * DEG_TO_RAD);

aiVector3D position = aiVector3D(100, 10, -20);

aiMatrix4x4 aMat = aiMatrix4x4(scaling, rotation, position);

importedScene->mRootNode->mTransformation = aMat;

````

Can you guys tell me what I'm doing wrong? Thanks

**Comment by kimkulling** [comment]
I can try. Which kind of issue do you get?

**Reply by kimkulling** [reply]
And aiMatrix4x4 has a constructor, which is not doing the correct matrix multiplications as far as I know.

**Reply by alessandroAM** [reply]
Hi Kim, as I mentioned, the above code should be supposed to set transformations for the mesh, but actually it does nothing. If, as you say, the aiMatrix4x4 is ineffective then that might be the reason.
But so, how do you actually apply transformations to an aiMesh? Cheers

**Reply by kimkulling** [reply]
To apply a transformation to a mesh you have to find the node which the mesh is assigned to. This reference is described by the node name and the area of meshes stores in aiNode. If you have found the node you can change the relative transformation in the node related to its root node. 
So if all meshes are related to the root node changing the transformation to the root shall affect all assigned meshes.

If this is not the case: check if any post-processing steps are active. You need to apply these transformations before doing the post-processing.

**Comment by Lucipetus** [comment]
AFAIK, your code sets the root node's mTransformation, and does nothing else.

How are you retrieving the mesh's data currently? Are you only using the mesh node's mTransformation? If that's the case, do know mTransformation is only relative to the parent of current node, you have to calculate the transform from root to any node yourself, and store them however you want.
